src/shared/ui/picker.ts

```diff
@@ -145,3 +146,92 @@ export function verifySinglePickerOutput<T extends vscode.QuickPickItem>(choices
+     */
+    public constructor(
+        private readonly awsCallLogic: {
+            // TODO: allow for creation of a new call in case we want to reload quick pick in its entirety
+            iteratingAwsCall: IteratingAWSCall<TRequest, TResponse>
+            initialRequest: TRequest
+            awsResponseToQuickPickItem: (response: TResponse) => vscode.QuickPickItem[]
+        },
+        private readonly pickerOptions: {
+            options?: vscode.QuickPickOptions & AdditionalQuickPickOptions
+            buttons?: vscode.QuickInputButton[]
+            onDidTriggerButton?: (
+                button: vscode.QuickInputButton,
+                resolve: (
+                    value: vscode.QuickPickItem[] | PromiseLike<vscode.QuickPickItem[] | undefined> | undefined
+                ) => void,
+                reject: (reason?: any) => void
+            ) => void
+        } = {}
+    ) {
+        // TODO: Create default buttons for load next page, refresh
+        // TODO: Set a global throttling flag that will optionally display said load next page button
+        this.quickPick = createQuickPick<vscode.QuickPickItem>({
+            options: {
+                ...this.pickerOptions.options,
+                onDidSelectItem: item => {
+                    // pause any existing execution
+                    this.isPaused = true
+                    // pass existing onDidSelectItem through if it exists
+                    if (this.pickerOptions.options?.onDidSelectItem) {
+                        this.pickerOptions.options.onDidSelectItem(item)
+                    }
+                },
+            },
+            items: this.items,
+            buttons: this.pickerOptions.buttons,
+        })
+
+        this.moreItemsRequest.event(() => this.loadItems())
+    }
+
+    /**
+     * Prompts the user with the quick pick specified by the constructor.
+     * Always attempts to load new results from the iteratingAwsCall, even if the call has been exhausted.
+     * If the call picker was previously paused, unpauses it.
+     */
+    public async promptUser(): Promise<vscode.QuickPickItem[] | undefined> {
+        // start background loading and unpause the loader (if it was paused previously by a selection)
+        this.quickPick.busy = true
+        this.isPaused = false
+        if (!this.isDone) {
+            this.moreItemsRequest.fire()
+        }
+        return await promptUser<vscode.QuickPickItem>({
+            picker: this.quickPick,
+            onDidTriggerButton: this.pickerOptions.onDidTriggerButton,
+        })
+    }
+
+    // TODO: Add nodes for no items, error (error retries call from where it left off?)
+    private async loadItems(): Promise<void> {
+        const iter = this.awsCallLogic.iteratingAwsCall.getIteratorForRequest(this.awsCallLogic.initialRequest)
+
+        for await (const item of iter) {
```

Comment:
`item` is also an iter/generator so I would use names like:

```suggestion
        for await (const responseIter of pageIter) {
```

Reply:
I don't think it is? It should be returning an item of TResponse type.

Reply:
yes,  I was confused about `AsyncGenerator<TResponse>` being the return type. That's just a typescript/js signature for an async generator, it returns TResponse. 🤢